extensions/wikia/AdEngine/src/ad-engine.bridge.js

```diff
@@ -0,0 +1,89 @@
+import Context from 'ad-engine/src/services/context-service';
+import SlotService from 'ad-engine/src/services/slot-service';
+import TemplateService from 'ad-engine/src/services/template-service';
+import BigFancyAdBelow from 'ad-products/src/modules/templates/uap/big-fancy-ad-below';
+import UniversalAdPackage from 'ad-products/src/modules/templates/uap/universal-ad-package';
+import StringBuilder from 'ad-engine/src/utils/string-builder';
+import config from './context';
+import Client from "ad-engine/src/utils/client";
+
+Context.extend(config);
+let supportedTemplates = [ BigFancyAdBelow ];
+
+supportedTemplates.forEach((template) => {
+	TemplateService.register(template);
+});
+
+function init(slotRegistry, pageLevelTargeting, skin) {
+	overrideSlotService(slotRegistry);
+	updatePageLevelTargeting(pageLevelTargeting, skin);
+}
+
+function overrideSlotService(slotRegistry) {
+	SlotService.getBySlotName = (id) => {
+		if (id) {
+			let slot = slotRegistry.get(id);
+			return unifySlotInterface(slot);
+		}
+	};
+}
+
+function unifySlotInterface(slot) {
+	const slotContext = Context.get(`slots.${slot.name}`) || {targeting: {}};
+	slot.getSlotName = () => slot.name;
+	slot.default = {
+		getSlotName: () => slot.name
+	};
+	slot.getId = () => slot.name;
+	slot.config = slotContext;
+	slot.getVideoAdUnit = () => buildVastAdUnit(slot.name);
+	slot.getTargeting = () => slotContext.targeting;
+	return slot;
+}
+
+function buildVastAdUnit(slotName) {
+	return StringBuilder.build(
+		Context.get(`vast.adUnitId`),
+		Object.assign({}, Context.get('targeting'), Context.get(`slots.${slotName}`))
+	);
+}
+
+
+function loadCustomAd(fallback) {
+	return (params) => {
+		if (getSupportedTemplateNames().includes(params.type)) {
+			const slot = SlotService.getBySlotName(params.slotName);
+			slot.container.parentNode.classList.add('gpt-ad');
+			Context.set(`slots.${params.slotName}.targeting.src`, params.src);
+			TemplateService.init(params.type, slot, params);
+		} else {
+			fallback(params);
+		}
+	};
+}
+
+function getSupportedTemplateNames() {
+	return supportedTemplates.map((template) => template.getName());
+}
+
+function updatePageLevelTargeting(params, skin) {
+	Context.set('custom.device', Client.getDeviceType());
+	Context.set('targeting.skin', skin);
+	Object.keys(params).forEach((key) => Context.set(`targeting.${key}`, params[key]));
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Convert this ES6 arrow function into a regular one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-js-rules%3AArrowFunctions)

[project-id]: wikia:app